shared/eip712/eip712.go

```diff
@@ -0,0 +1,84 @@
+package eip712
+
+import (
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/accounts"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+)
+
+type EIP712Components struct {
+	R [32]byte `json:"r"`
+	S [32]byte `json:"s"`
+	V uint8    `json:"v"`
+}
+
+const EIP712Length = 65
+
+// Pretty print for EIP712Components
+func (e *EIP712Components) Print() {
+	fmt.Println("EIP712 Components:")
+	fmt.Printf("R: %x\n", e.R)
+	fmt.Printf("S: %x\n", e.S)
+	fmt.Printf("V: %d\n", e.V)
+}
+
+// UnmarshallText verifies the length of a decoded EIP-712 signature and assigns the appropriate bytes to R/S/V
+func (e *EIP712Components) UnmarshallText(inp []byte) error {
+	if len(inp) != EIP712Length {
+		return fmt.Errorf("error decoding EIP-712 signature string: invalid length %d bytes (expected %d bytes)", len(inp), EIP712Length)
+	}
+
+	copy(e.R[:], inp[0:32])
+	copy(e.S[:], inp[32:64])
+	e.V = inp[64]
+
+	return nil
+}
+
+// MarshallText initializes an empty byte slice, copies fields R/S/V into signatureBytes then returns it
+func (e *EIP712Components) MarshallText() ([]byte, error) {
```

Comment:
`MarshalText`

function name is actually super important, otherwise we aren't implementing the TextMarshaler interface.